autoload/black.vim

```diff
@@ -51,39 +53,52 @@ def _get_python_binary(exec_prefix, pyver):
   exec_path = (bin_path / f"python3").resolve()
   if exec_path.exists():
     return exec_path
+  # TODO: Instead of failing, try to find black in PATH first and use its
+  # exec_prefix. This can happen when black is installed systemwide and
+  # g:black_virtualenv was not updated.
   raise ValueError("python executable not found")
 
+def _get_python_version(exec_path, as_string=False):
+  import subprocess
+  version_proc = subprocess.run([exec_path, "--version"], stdout=subprocess.PIPE, text=True)
+  version = version_proc.stdout.split(" ")[1].strip()
+  if not as_string:
+    version = tuple(map(int, version.split(".")))
+  return version
+
 def _get_pip(venv_path):
   if sys.platform[:3] == "win":
     return venv_path / 'Scripts' / 'pip.exe'
   return venv_path / 'bin' / 'pip'
 
-def _get_virtualenv_site_packages(venv_path, pyver):
+def _get_virtualenv_site_packages(venv_path):
   if sys.platform[:3] == "win":
     return venv_path / 'Lib' / 'site-packages'
-  return venv_path / 'lib' / f'python{pyver[0]}.{pyver[1]}' / 'site-packages'
+  venv_python_version = _get_python_version(_get_python_binary(venv_path))
+  return venv_path / 'lib' / f'python{venv_python_version[0]}.{venv_python_version[1]}' / 'site-packages'
 
 def _initialize_black_env(upgrade=False):
+  virtualenv_path = Path(vim.eval("g:black_virtualenv")).expanduser()
+  virtualenv_site_packages = str(_get_virtualenv_site_packages(virtualenv_path))
   if vim.eval("g:black_use_virtualenv ? 'true' : 'false'") == "false":
     if upgrade:
       print("Upgrade disabled due to g:black_use_virtualenv being disabled.")
       print("Either use your system package manager (or pip) to upgrade black separately,")
       print("or modify your vimrc to have 'let g:black_use_virtualenv = 1'.")
       return False
     else:
-      # Nothing needed to be done.
+      if virtualenv_site_packages not in sys.path:
+        sys.path.insert(0, virtualenv_site_packages)
       return True
 
+  # TODO: Is this check about the Vim plugin, or about Black?
```

Comment:
It's about Black primarily; Black itself requires 3.8+ (since Python 3.7 is no longer supported). By default I would want the same version requirement for the Vim plugin, but if there's some reason to keep supporting older versions in the plugin code, I'm open to it.